File: 1923-sentence-similarity-iii/sentence-similarity-iii.cpp
class Solution {
public:
    bool areSentencesSimilar(string s1, string s2) {
        if(s1.size()<s2.size())
        swap(s1,s2);

        vector<string>w1,w2;

       
        stringstream ss(s1);
         string word;
         while(ss>>word)
         {
            w1.push_back(word);
         }

 stringstream ss1(s2);
        
         while(ss1>>word)
         {
            w2.push_back(word);
         }

         int i=0;
         int l=0;
         int j=w1.size()-1;
         int r=w2.size()-1;
         while(i<w1.size() &&l<w2.size() &&w1[i]==w2[l])
         {
            i++;
            l++;
         }
           while(j<w1.size() &&r<w2.size() &&w1[j]==w2[r])
         {
            j--;
            r--;
         }

         return (r<l);






    }
};

File: 1923-sentence-similarity-iii/sentence-similarity-iii.py
class Solution:
    def areSentencesSimilar(self, s1: str, s2: str) -> bool:

        arr1 = s1.split()
        arr2 = s2.split()

        if len(arr1) < len(arr2):
            arr1, arr2 = arr2, arr1

        i = 0
        j = 0

        while i < len(arr1) and j < len(arr2):
            if arr1[i] == arr2[j]:
                i += 1
                j += 1
            else:
                break

        i1 = len(arr1) - 1
        j1 = len(arr2) - 1

        while i1 >= 0 and j1 >= 0:
            if arr1[i1] == arr2[j1]:
                i1 -= 1
                j1 -= 1
            else:
                break

        return j > j1